File: IQ/Classes/clsRole.vb
Imports dataAccess

Public Class clsRole

    Property ID As Integer
    Property Code As String
    Property Translation As clsTranslation
    Property Rights As Dictionary(Of Integer, clsRight)
    Property i_right_code As Dictionary(Of String, clsRight)


    ReadOnly Property DisplayName(language As clsLanguage)
        Get
            DisplayName = Me.Translation.text(language)
        End Get
    End Property

    ReadOnly Property EnglishDisplayName
        Get
            EnglishDisplayName = Me.Translation.text(English)
        End Get
    End Property

    Public Sub New()
        Me.ID = -1
        Me.Rights = New Dictionary(Of Integer, clsRight)
        Me.i_right_code = New Dictionary(Of String, clsRight)()

    End Sub
    Public Sub New(Code As String, translation As clsTranslation)

        Dim sql$
        sql$ = "INSERT INTO [Role] (code,fk_Translation_key) "
        sql$ &= " values (" & da.SqlEncode(Code) & "," & translation.Key & ");"

        Me.ID = da.DBExecutesql(sql$, True)
        Me.Code = Code
        Me.Translation = translation
        Me.Rights = New Dictionary(Of Integer, clsRight)
        Me.i_right_code = New Dictionary(Of String, clsRight)()

        iq.i_role_Code.Add(Me.Code, Me)

    End Sub

    Public Sub New(Id As Integer, Code As String, translation As clsTranslation)

        Me.ID = Id
        Me.Code = Code
        Me.Translation = translation
        Me.Rights = New Dictionary(Of Integer, clsRight)
        Me.i_right_code = New Dictionary(Of String, clsRight)()

        iq.i_role_Code.Add(Me.Code, Me)

    End Sub

    Public Sub AddRight(right As clsRight)
        If Rights.ContainsKey(right.ID) Then Exit Sub
        Dim sql$
        sql$ = "INSERT INTO [RoleRight] (fk_Role_Id,fk_right_id) "
        sql$ &= " values (" & Me.ID & "," & right.ID & ");"

        da.DBExecutesql(sql$, True)
        Me.Rights.Add(right.ID, right)
        Me.i_right_code.Add(right.Code, right)
    End Sub

End Class 'clsRole

File: IQ/Classes/clsRole.cs
using dataAccess;


public class clsRole
{

    public int ID { get; set; }
    public string Code { get; set; }
    public clsTranslation Translation { get; set; }
    public Dictionary<int, clsRight> Rights { get; set; }
    public Dictionary<string, clsRight> i_right_code { get; set; }


    public dynamic get_DisplayName(clsLanguage language)
    {
        dynamic returnValue = default(dynamic);
        returnValue = this.Translation.text(language);
        return returnValue;
    }

    public dynamic EnglishDisplayName
    {
        get
        {
            dynamic returnValue = default(dynamic);
            returnValue = this.Translation.text(English);
            return returnValue;
        }
    }

    public clsRole()
    {
        this.ID = -1;
        this.Rights = new Dictionary<int, clsRight>();
        this.i_right_code = new Dictionary<string, clsRight>();

    }
    public clsRole(string Code, clsTranslation translation)
    {

        object sql = null;
        sql = "INSERT INTO [Role] (code,fk_Translation_key) ";
        sql += " values (" + da.SqlEncode(Code) + "," + Translation.Key + ");";

        this.ID = System.Convert.ToInt32(da.DBExecutesql(sql, true));
        this.Code = Code;
        this.Translation = translation;
        this.Rights = new Dictionary<int, clsRight>();
        this.i_right_code = new Dictionary<string, clsRight>();

        iq.i_role_Code.Add(this.Code, this);

    }

    public clsRole(int Id, string Code, clsTranslation translation)
    {

        this.ID = Id;
        this.Code = Code;
        this.Translation = translation;
        this.Rights = new Dictionary<int, clsRight>();
        this.i_right_code = new Dictionary<string, clsRight>();

        iq.i_role_Code.Add(this.Code, this);

    }

    public void AddRight(clsRight right)
    {
        if (Rights.ContainsKey(right.ID))
        {
            return;
        }
        object sql = null;
        sql = "INSERT INTO [RoleRight] (fk_Role_Id,fk_right_id) ";
        sql += " values (" + System.Convert.ToString(this.ID) + "," + right.ID + ");";

        da.DBExecutesql(sql, true);
        this.Rights.Add(right.ID, right);
        this.i_right_code.Add(right.Code, right);
    }

} //clsRole